plugins/cad/src/components/RepositoryPage/components/AdvancedRepositoryOptions.tsx

```diff
@@ -41,10 +48,33 @@ export const AdvancedRepositoryOptions = () => {
   };
 
   const executeUnregisterRepository = async (): Promise<void> => {
-    await api.unregisterRepository(repositoryName);
+    const repoSecretName =
+      repositorySummary.repository.spec.git?.secretRef?.name;
+    const repoNamespace = 
+      repositorySummary.repository.metadata.namespace;
+
+    if(repoSecretName && !checkUsedByOtherRepository(repoSecretName)) {
+      await Promise.all([api.unregisterRepository(repositoryName), api.deleteSecret(repoSecretName, repoNamespace)]);
+    } else {
+      await api.unregisterRepository(repositoryName);
+    }
     navigate(repositoriesRef());
   };
 
+  const checkUsedByOtherRepository = async (secretName : string): Promise<boolean> => {
```

Comment:
I don't know if this is the most efficient way of doing this...

Reply:
The request to `listRepositories` make sense since we'll need to see if the secret is shared with multiple repositories. 

The function at the moment only returns true if the secret is shared with one other repository, and false if it shared more than one other repository (for example, with 3 other repositories), so this will need to be fixed.

You can potentially update the code to the following: 
```
  const checkUsedByOtherRepository = async (secretName : string): Promise<boolean> => {
    const { items: repositories } = await api.listRepositories();

    const isSecretShared = repositories.some(repository => repository.metadata.name !== repositoryName && repository.spec.git?.secretRef?.name === secretName)

    return isSecretShared;
  }
```

This takes advantage of [Array.prototype.some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) letting us know if another repository is sharing the secret without needing to keep a count.